bndtools.builder/src/org/bndtools/builder/facade/IProjectBuilder.java

```diff
@@ -0,0 +1,78 @@
+package org.bndtools.builder.facade;
+
+import java.util.Map;
+
+import org.eclipse.core.resources.IBuildConfiguration;
+import org.eclipse.core.resources.IBuildContext;
+import org.eclipse.core.resources.ICommand;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResourceDelta;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.jobs.ISchedulingRule;
+
+public abstract class IProjectBuilder {
```

Comment:
:+1: Originally it was an interface and then I realised I needed to store some state and needed the abstract class. Perhaps I will use "IncrementalProjectBuilderDelegate".

But I'm not the first one to use the "I" naming scheme in Bndtools - there are a few pre-existing in org.bndtools.api. I also added `ICentral` in the very first commit. I understand that Eclipse can tell you if it's a type or an interface, but usually this is not quite as convenient as the "I" convention which you can easily see at a glance in the source without having to do any kind of inspection (unless, of course, someone does something silly like I did here which violates the convention... :smile:). I'm not fussed either way - happy to avoid using the "I" convention, but with "Central" I needed something different to distinguish it from the concrete class from which I extracted it originally.

Reply:
Renamed it ProjectBuilderDelegate